projects/hr-portal/src/hr-portal.ts

```diff
@@ -1,83 +1,265 @@
+    });
   }
 
+  connectedCallback() {
+    super.connectedCallback();
+    this.dataService.fetchData().then((data) => {
+      this.data = data;
+    });
+  }
 
+  @query("igc-tree-grid")
+  private treeGrid!: IgcTreeGridComponent;
 
-  static styles = css`
-    :host {
-      max-width: 1280px;
-      margin: 0 auto;
-      padding: 2rem;
-      text-align: center;
+  clearSorting() {
+    if (this.treeGrid) {
+      this.treeGrid.sortingExpressions = [];
     }
+    this.isSorted = false;
+  }
 
-    .logo {
-      height: 6em;
-      padding: 1.5em;
-      will-change: filter;
-      transition: filter 300ms;
+  handleSortingChanged() {
+    if (this.treeGrid) {
+      this.isSorted = this.treeGrid.sortingExpressions.length > 0;
+      console.log(
+        "Sorting expressions changed:",
+        this.treeGrid.sortingExpressions
+      );
     }
-    .logo:hover {
-      filter: drop-shadow(0 0 2em #646cffaa);
-    }
-    .logo.lit:hover {
-      filter: drop-shadow(0 0 2em #325cffaa);
+  }
+
+  public avatarTemplate = (ctx: IgcCellTemplateContext) => {
+    let row = ctx.cell.row;
+    return html`
+      <div class="employeeDiv">
+        <igc-avatar shape="rounded" .src="${row.data.Picture}"></igc-avatar>
+        <span>${row.data.Name}</span>
+      </div>
+    `;
+  };
+
+  public countryIconTemplate = (ctx: IgcCellTemplateContext) => {
+    let row = ctx.cell.row;
+    return html` <div class="flagDiv">
+      <igc-icon
+        collection="country-icons"
+        name="${row.data.Country}"
+      ></igc-icon>
+      <span>${row.data.Location}, ${row.data.Country}</span>
+    </div>`;
+  };
+
+  public contactsTemplate = (ctx: IgcCellTemplateContext) => {
+    let row = ctx.cell.row;
+    return html`
+      <div class="center-content">
+        <a href="mailto:${row.data.Email}">
+          <igc-icon-button
+            collection="hr-icons"
+            name="mail"
+            class="small"
+            variant="flat"
+          ></igc-icon-button>
+        </a>
+        <a href="tel:${row.data.Phone}">
+          <igc-icon-button
+            collection="hr-icons"
+            name="tel"
+            class="small"
+            variant="flat"
+          ></igc-icon-button>
+        </a>
+        <a href="https://www.linkedin.com" target="_blank">
+          <igc-icon-button
+            collection="hr-icons"
+            name="linkedIn"
+            class="small"
+            variant="flat"
+          ></igc-icon-button>
+        </a>
+      </div>
+    `;
+  };
+
+  public dateTemplate = (ctx: IgcCellTemplateContext) => {
+    let row = ctx.cell.row;
+    const date = new Date(row.data.HireDate);
+    const formattedDate = date.toISOString().split("T")[0];
+    return html` ${formattedDate} `;
+  };
+
+  render() {
+    return html`
+      <link rel="stylesheet" href="node_modules/igniteui-webcomponents-grids/grids/themes/light/fluent.css" />
+      <div class="ig-typography rootDiv">
```

Comment:
Why do you need that div container?

Reply:
the typography does not apply otherwise even tho its already applied to the body

Reply:
It does on my machine. Are you sure it doesnt?

Reply:
yes, it came as suggestion from @skrustev  as he said it didn't on his as well